test/integration_test.go

```diff
@@ -2425,6 +2425,32 @@ func (ts *IntegrationTestSuite) registerNamespace() {
 	}
 }
 
+func (ts *IntegrationTestSuite) TestHistoryLength() {
```

Comment:
To get some extra coverage, I'd evict this workflow from cache (fail a task?) to verify that the history length is correct during replay too.

Reply:
Maybe I can complete the workflow then run a query for the same value I'm returning

Reply:
Note, our integration tests run in no-cache mode too

Reply:
That should be enough then, I'd just maybe mention that we get the coverage from that

Reply:
Definitely add some tests using local activities with and without replay too, per the ordering thing I mention in my other comment

Reply:
Updated the integration test to also use local activities and added comment making it clear we use the zero-cache integration test to confirm replay (and have also manually confirmed that the zero-cache does replay properly).